src/dbnode/storage/block/block.go

```diff
@@ -263,13 +263,6 @@ func (b *dbBlock) HasMergeTarget() bool {
 	return hasMergeTarget
 }
 
-func (b *dbBlock) IsRetrieved() bool {
-	b.RLock()
-	retrieved := b.retriever == nil
-	b.RUnlock()
-	return retrieved
-}
-
 func (b *dbBlock) WasRetrievedFromDisk() bool {
```

Comment:
Another random thing, I think we might be able to delete the `OnRetrieveBlock` method in the DatabaseBlock. I think its primary purpose was to allow us to separate the following two actions:

1) Resetting a block
and
2) Fetching the blocks data

Without the CacheAllMetadata policy, these two steps will never be separate so you can probably combine it all into one function call. For example in the series we do this:

```golang
b.ResetRetrievable(startTime, blockSize, s.blockRetriever, metadata)
	// Use s.id instead of id here, because id is finalized by the context whereas
	// we rely on the G.C to reclaim s.id. This is important because the block will
	// hold onto the id ref, and (if the LRU caching policy is enabled) the shard
	// will need it later when the WiredList calls its OnEvictedFromWiredList method.
	// Also note that ResetRetrievable will mark the block as not retrieved from disk,
	// but OnRetrieveBlock will then properly mark it as retrieved from disk so subsequent
	// calls to WasRetrievedFromDisk will return true.
	b.OnRetrieveBlock(s.id, tags, startTime, segment)
```

It seems like that could just become

```golang
b.Reset(startTime, blockSize, segment)
````

And then we could delete the `OnRetrieveBlock` and `ResetRetrievable` methods (and their helper methods like `resetRetrievableWithLock`) since they're only required in a world where data can be lazily fetched

Reply:
@robskillington